frontend/src/modules/app/ToolBar.tsx

```diff
@@ -75,7 +75,34 @@ const useStyles = makeStyles((theme: Theme) => ({
  */
 export const HeaderClock = (): JSX.Element => {
   const classes = useStyles();
-  const clockTime = useSelector(getClockTime);
+  // we can initialize with something more specific like wall clock time
+  const [clockTime, setClockTime] = useState(
+    new Date().toLocaleTimeString([], {
+      hour: '2-digit',
+      minute: '2-digit',
+      second: '2-digit',
+      hour12: true,
+    }),
+  );
+
+  /**
+   * UseEffect to run the clock every second
+   */
+  useEffect(() => {
+    const clockTimer = setInterval(() => {
+      setClockTime(
+        new Date().toLocaleTimeString([], {
+          hour: '2-digit',
+          minute: '2-digit',
+          second: '2-digit',
+          hour12: true,
+        }),
+      );
+    }, 1000);
+    return () => {
```

Comment:
This is really interesting, (when) is useEffect able to take advantage of the function you return? If so, I believe we may be able to use this in a few other places in the frontend too.

Reply:
So this is sort of cleanup, return is more like when the component is unmounted we should clear the timer, I believe in React class components we have a constructor by a similar name, so return does the same thing.

Reply:
I see - that would be good to keep in mind anywhere else we allocate resources (such as `setInterval`s) inside `useEffect`s.